### DISCUSSION TITLE
Sometimes, component doesn't render without any console errors

### DISCUSSION THREAD

**Discussion by realpratiknikam** [add2cal/add-to-calendar-button#646] [Q&A]
Is there a known issue where component doesn't render without any console errors? This is really hard to reproduce so not able to get to bottom of it.

- Every component prop is current. In-fact it doesn't even change so it's not an issue with props being passed.
- Have seen it fail mostly on mobile devices.

**Comment by jekuer** [comment]
Can you share your config, please?
Could help to get closer to the root of this.
And also how you are including it. Are you, for example, adding props dynamically? Could be a race condition then. The button should catch any changes, but depending on your setup/framework, there could be some edge cases.

You could also add the debug option to get some more information. If it does not show anything with the debug option set, I would rather assume it has something to do with the surrounding setup.

**Comment by realpratiknikam** [comment]
Sure, here's how it looks like. Implementation is pretty straightforward with dynamic prop values which would always present.

I've added debug prop now and not able to reproduce so far so let's see if it does expose something.

```
          <AddToCalendarButton
            name={event?.title}
            description={event?.description}
            startDate={formatStartDateTimeToIso(
              event?.date,
              event?.startTime,
              event?.timezone
            )}
            endDate={formatEndDateTimeToIso(
              event?.date,
              event?.endTime,
              event?.timezone
            )}
            location={event?.venue}
            timeZone={event?.timezone?.value}
            options={["Apple", "Google", "Yahoo", "iCal"]}
            listStyle="overlay"
            buttonsList
            size="4"
          />
          ```

**Comment by realpratiknikam** [comment]
No luck so far with reproduction :(
